api/client/webclient.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2020-2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/trace"
+)
+
+const (
+	invalidProxyAddressMessage = "'%v' is not a valid proxy address"
+)
+
+// initClient creates a new client to the HTTPS web proxy.
+func initClient(proxyAddr string) (*http.Client, error) {
```

Comment:
Since `NewInsecureWebClient` is not used anywhere outside of this package, rename `initClient` to `newWebClient` and inline `NewInsecureWebClient` in it